### ISSUE TITLE
ly starts at wrong resolution and gives me an error if i login too fast

### ISSUE THREAD

**Issue by Hashino** [fairyglade/ly#428]
every time i boot my system ly starts at a wrong (smaller) resolution than my native one. some seconds after it tries to resize itself, more often than not with no success and ends up completely distorted. it fixes itself after i start to type the password.

another problem is that if type my password too fast (before the backlight config is loaded) X gives me an error "no screens found"

**Comment by AnErrupTion** [comment]
> every time i boot my system ly starts at a wrong (smaller) resolution
Sadly I don't think we can fix that universally, as t here is no global way to control the TTY in that way. Also, do you have 2 or more monitors?

> every time i boot my system ly starts at a wrong (smaller) resolution
Hmm... I'd like to see how that looks like.

> another problem is that if type my password too fast (before the backlight config is loaded) X gives me an error "no screens found"
The backlight config? What do you mean by that?

**Comment by Hashino** [comment]
> Sadly I don't think we can fix that universally, as t here is no global way to control the TTY in that way. Also, do you have 2 or more monitors?

nope, just one, its a laptop.

>Hmm... I'd like to see how that looks like.

[here's a video](https://www.youtube.com/watch?v=bI2nVfnn3Bs)

>The backlight config? What do you mean by that?

you can see at the end of the video the screen brightness lowering. if i try to login before that i get a "no screens found" error by X
i tried adding `After="systemd-backlight@backlight:radeon_bl0.service"` to `/etc/systemd/system/display-manager.service` but it didn't do anything

**Comment by AnErrupTion** [comment]
> nope, just one, its a laptop.

Then forget what I said, it should resize itself automatically.

> [here's a video](https://www.youtube.com/watch?v=bI2nVfnn3Bs)

Oh dear. I don't know what is happening, that's extremely weird for sure.

> but it didn't do anything

Yeah, well with how broken ly looks, I'm not surprised. Just to confirm, can you test going to a regular TTY and see if everything shows correctly?

**Comment by Hashino** [comment]
yes, it works fine when i go to another tty. also, i forgot to show it in the video, but once i start typing the password ly fixes itself

**Comment by Hashino** [comment]
is there a way to delay ly so i don't get the "no screens found" error by X? looking at [systemd-backlight@backlight:radeon_bl0.service](https://www.freedesktop.org/software/systemd/man/systemd-backlight@.service.html) description it seems to be the service that load the screen brightness but  adding `After="systemd-backlight@backlight:radeon_bl0.service"` to `/etc/systemd/system/display-manager.service` didn't achieve anything.

**Comment by AnErrupTion** [comment]
I'm not sure, sorry. Maybe @nullgemm or someone else could help on that.

**Comment by YanDoroshenko** [comment]
I have a very similar issue - if run on a single laptop internal display, ly input window is displayed in the top left corner. After you hit the first character of the password, it jumps to the center. Not the end of any particular worlds, but a bit annoying. With multiple monitors it's always in the center.

My guess would be that it has something to do with video driver initialization, but what do I know.
